dev/com.ibm.ws.cdi.1.2.weld/src/com/ibm/ws/cdi/proxy/ProxyServicesImpl.java

```diff
@@ -53,58 +54,43 @@ public void cleanup() {
-                // not be able to know exactly which bundles are providing bean
-                // types until this getClassLoader method is called.  But by the time
-                // this method is called it is too late for a weaving hook to do
-                // anything because weld is going to use the returned class loader
-                // immediately to reflectively define a proxy class.  The class loader
-                // MUST have visibility to the weld packages before this reflective
-                // call to defineClass.
-                ClassLoader cl = proxiedBeanType.getClassLoader();
-                if (cl == null) {
-                    cl = CLASS_LOADER_FOR_SYSTEM_CLASSES;
-                } else if (cl instanceof BundleReference) {
-                    Bundle b = ((BundleReference) cl).getBundle();
-                    addWeldDynamicImports(b, WELD_PACKAGES);
-                }
-                return cl;
-            }
-        });
+        // This makes it impossible to use weaving hooks to add new dynamic
+        // import packages.  The weaving hook approach requires
+        // a weaving hook registration that knows ahead of time what
+        // bundles provide CDI bean types and then on first class define using
+        // that bundle's class loader the weaving hook would add the necessary
+        // weld packages as dynamic imports.  We cannot and will
+        // not be able to know exactly which bundles are providing bean
+        // types until this getClassLoader method is called.  But by the time
+        // this method is called it is too late for a weaving hook to do
+        // anything because weld is going to use the returned class loader
+        // immediately to reflectively define a proxy class.  The class loader
+        // MUST have visibility to the weld packages before this reflective
+        // call to defineClass.
+        ClassLoader cl = proxiedBeanType.getClassLoader();
+        if (cl == null) {
+            cl = CLASS_LOADER_FOR_SYSTEM_CLASSES;
+        } else if (cl instanceof BundleReference) {
+            Bundle b = ((BundleReference) cl).getBundle();
+            addWeldDynamicImports(b, WELD_PACKAGES);
+        }
+            return cl;
     }
-
+    
     @Override
     public Class<?> loadBeanClass(final String className) {
-        //This is tricky. Sometimes we need to use app classloader to load some app class
-        try {
-            return (Class<?>) AccessController.doPrivileged(new PrivilegedExceptionAction<Object>() {
-                @Override
-                public Object run() throws Exception {
-                    return Class.forName(className, true, getClassLoader(this.getClass()));
-                }
-            });
-        } catch (PrivilegedActionException pae) {
-            throw new CDIRuntimeException(pae.getException());
-        }
+        throw new UnsupportedOperationException("This method is not implemented");
     }
```

Comment:
It seems odd to change this version to throw `UnsupportedOperationException` but in the 2.0 version we removed the `doPrivileged` call.

I think I'd lean towards just removing the `doPrivileged` call in both, but I'd expect them to be the same unless there's some reason why they shouldn't be.